### PR TITLE
Auth development

### PR DESCRIPTION
**Pull request by Rubinstd**

### CODE REVIEW

File: api/common/controller/TriageController.py [hunk 1]
```diff
@@ -29,11 +31,11 @@ class TriageController:
 
     # Database connection information
     DATABASE_DATA = {
-        'database': 'triage',
-        'user': 'triage_controller',
-        'password': 'password',
-        'host': 'db',
-        'port': '5432'
+        'user': 'admin',
+        'password': 'docker',
```
Comment (SEsseltine) [inline api/common/controller/TriageController.py:35]: Should these be admin or should the perms of triage_controller be updated?

File: api/common/ClinicData.py [hunk 2]
```diff
@@ -23,11 +22,11 @@ class ClinicData:
 
     # Database connection information
     DATABASE_DATA = {
-        'database': 'triage',
         'user': 'admin',
-        'password': 'password',
-        'host': 'db',
-        'port': '5432'
+        'password': 'docker',
```
Comment (SEsseltine) [inline api/common/ClinicData.py:26]: Do we need a new user here to handle this work?

File: api/resources/auth.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+SECRET_KEY = 'thisisthesecretkey'
+
+class Auth(Resource):
+    """
+    The `Predict` class handles all of the requests relative to Prediction for the API.
+    """
+
+    DATABASE_DATA = {
+        'user': 'admin',
+        'password': 'docker',
```
Comment (SEsseltine) [inline api/resources/auth.py:28]: Let's add the user for this in the db before merge

File: api/resources/auth.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+# Internal dependencies
+from api.common.database_interaction import DataBase
+from api.common.config import database_config
+
+SECRET_KEY = 'thisisthesecretkey'
+
+class Auth(Resource):
+    """
+    The `Predict` class handles all of the requests relative to Prediction for the API.
```
Comment (SEsseltine) [inline api/resources/auth.py:23]: Let's update the commenting here to reflect the proper class

File: api/resources/auth.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    The required schema to handle a get request
+
+    Args:
+        clinic_id (int): The id of the clinic being referenced.
+        start_date (str): The start date of the predictions.
+        end_date (str): The end date of the predictions.
+        intervals (list): Date intervals for predictions to be grouped by.
+        confidence (float): Required prediction confidence.
+        num_sim_runs (int): Number of simulations to run.
+        waitlist (file): Current wait list of patients for the clinic.
```
Comment (SEsseltine) [inline api/resources/auth.py:54]: Let's correct the arguments here for this endpoint

File: api/resources/auth.py [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+            }
+            ```
+
+        """
+        # Validate input arguments.
+        args = parser.parse(self.arg_schema_post, request, location='json')
+        
+        user_clinic = self.__validate_user(args['username'], args['password'])
+
+        return json.dumps({ 'token': self.__generate_token(args['username'], user_clinic), 'clinic_id': user_clinic })
```
Comment (SEsseltine) [inline api/resources/auth.py:85]: Are we still planning on returning if the user is an admin?

File: api/resources/auth.py [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+            
+
+    def __validate_user(self, username, password):
+        db = DataBase(self.DATABASE_DATA)
+        query = "SELECT clinic_id FROM triagedata.users "
```
Comment (SEsseltine) [inline api/resources/auth.py:90]: did we want to use the salt value here?

File: api/resources/models.py [hunk 8]
```diff
@@ -1,29 +1,29 @@
...
-
-class Models(Resource):
+class Models(AuthResource):
     """
     The `Models` class handles all of the requests relative to Models
     for the API.
     """
     DATABASE_DATA = {
-        'database': 'triage',
         'user': 'model_handler',
```
Comment (SEsseltine) [inline api/resources/models.py:23]: We need to re-add password here or this should break

**Review by SEsseltine — CHANGES_REQUESTED** [review]

File: api/resources/auth.py (hunk 6 above)
Comment (Rubinstd) [reply api/resources/auth.py]: Yes, this has been added

File: api/resources/auth.py (hunk 7 above)
Comment (Rubinstd) [reply api/resources/auth.py]: Yes, this has been added now

**Review by SEsseltine — APPROVED** [review]

**Pull request merged**